field/TinyLang/Field/Evaluator.hs

```diff
@@ -144,57 +187,75 @@ evalBinOp Mul x y = return . UniConst Field $ x `mul` y
 evalBinOp Div x y = UniConst Field <$> divEval x y
 evalBinOp BAt x y = asIntegerEval x >>= fmap (UniConst Bool) . flip atEval y . fromIntegral
 
-evalStatementUni
-    :: (MonadEvalError f m, Eq f, Field f, AsInteger f)
-    => Env (SomeUniConst f) -> Statement f -> Expr f a -> m (UniConst f a)
-evalStatementUni env (ELet (UniVar _ var) def) rest = do
-    defR <- evalExprUni env def
-    evalExprUni (insertVar var (Some defR) env) rest
-evalStatementUni env (EAssert expr) rest = do
-    exprR <- evalExpr env expr
-    if exprR
-        then evalExprUni env rest
-        else throwError $ AssertionFailedEvalError expr
+evalProgramUni :: (Monad m, Eq f, Field f, AsInteger f)
+    => Program f -> EvalT f m ()
+evalProgramUni = evalStatementsUni . unProgram
+
+evalStatementsUni :: (Monad m, Eq f, Field f, AsInteger f)
```

Comment:
@effectfully, I refactored the evaluator to a new transformer stack, per our discussion.

I was wondering if this method could be improved by using CPS.

Also, I will probably remove the `Statements f` wrapper, as it does not help anymore.

Reply:
Done.